File: src/main/java/com/chatchatabc/jpademojava/domain/model/Role.java
package com.chatchatabc.jpademojava.domain.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.security.core.GrantedAuthority;

import java.util.List;

@Data
@Entity
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "roles")
public class Role implements GrantedAuthority {
    @JsonIgnore
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;

    @JsonIgnore
    @Column(unique = true)
    private String name;

    @JsonIgnore
    @ManyToMany(mappedBy = "roles")
    private List<User> users;

    /**
     * Get the authority name
     *
     * @return
     */
    @Override
    public String getAuthority() {
        return this.name;
    }
}

File: src/main/kotlin/com/chatchatabc/jpademo/domain/model/Role.kt
package com.chatchatabc.jpademo.domain.model

import com.fasterxml.jackson.annotation.JsonIgnore
import jakarta.persistence.*
import lombok.Data
import org.springframework.security.core.GrantedAuthority

@Data
@Entity
@Table(name = "roles")
open class Role(name: String) : GrantedAuthority {

    @JsonIgnore
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    open lateinit var id: String

    @JsonIgnore
    @Column(unique = true)
    open var name: String = name

    @JsonIgnore
    @ManyToMany(mappedBy = "roles")
    open var users: MutableList<User> = mutableListOf()

    /**
     * Get Authority
     */
    override fun getAuthority(): String {
        return this.name
    }
}